src/bot/Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM python:3.10
+
+WORKDIR /code
+
+COPY ./requirements.txt requirements.txt
+
+RUN pip install --no-cache-dir --upgrade -r /code/requirements.txt
+
+COPY ./src ./src
```

Comment:
We're going to build each app's docker image from it's respective folder (i.e. `src/webapp` and `src/bot`).

When building a docker image there is the concept of a build context, which is generally the folder you run the build command from - here is a little break down:
```sh
docker build . -t myimage
```
* `docker`: the main command used to interact with the docker command-line interface (CLI).
* `build`: this is a subcommand of docker, which tells docker to build a new image from the provided build context. The build context is a directory containing the necessary files needed to create the image (i.e. `Dockerfile`). During the build process
* `.`: this is the build context, (the first argument to the build subcommand) which in this case is the current directory. Docker will look for a Dockerfile in the build context and use it to build the image. You could also put in a different directory (other than the current dir) as long as it has a Dockerfile in it.
* `-t myimage`: the tag for the image - basically just the name. These are usually in the format `name:version` (i.e. `myimage:0.0.1`), if you don't have a `:version` part of the tag, `:latest` is automatically added. So the result here would be `myimage:latest`

So for our situation here, you could either build the image with:
```sh
cd src/bot
docker build . -t yeelight_bot
```
or 
```sh
docker build ./src/bot -t yeelight_bot
```

And you can only `COPY` files from inside the build context (in this case `src/bot`). Since the `src` folder isn't in the `src/bot` directory (it's one of it's parents) you're going to have to adjust it so that it copies over files from the build context

Reply:
Fixed